OrbitCore/CoreApp.h

```diff
@@ -45,11 +50,26 @@ class CoreApp {
                          const std::string& /*a_Module*/,
                          const std::string& /*a_Name*/) {}
   virtual void AddKeyAndString(uint64_t key, std::string_view str) {}
+  virtual bool SelectProcess(const std::string& a_Process) { return false; }
+  virtual void OnRemoteModuleDebugInfo(const std::vector<ModuleDebugInfo>&) {}
+  virtual void ApplySession(std::shared_ptr<Session> session) {};
   virtual const std::unordered_map<DWORD64, std::shared_ptr<class Rule> >*
   GetRules() {
     return nullptr;
   }
   virtual void RefreshCaptureView() {}
+
+  // Managers
+  std::shared_ptr<orbit::TransactionManager> GetTransactionManager() {
+    return transaction_manager_;
+  }
+  std::shared_ptr<orbit::SymbolsManager> GetSymbolsManager() {
+    return symbols_manager_;
+  }
+
+  private:
+    std::shared_ptr<orbit::TransactionManager> transaction_manager_ = nullptr;
+    std::shared_ptr<orbit::SymbolsManager> symbols_manager_ = nullptr;
 };
 
-extern CoreApp* GCoreApp;
+extern std::shared_ptr<CoreApp> GCoreApp;
```

Comment:
see my comment about smart pointers and owning of resources.

Reply:
Changed it back to unique_ptr for OrbitApp and raw pointer for CoreApp.